app/services/versioned_files_service/purge_content_action.rb

```diff
@@ -0,0 +1,34 @@
+class VersionedFilesService
+  # Deletes content files that aren't referenced by any cocina version files.
+  class PurgeContentAction
+    # @param service [VersionedFilesService] the service
+    def initialize(service:)
+      @service = service
+    end
+
+    def call
+      purge_content_files
+    end
+
+    private
+
+    delegate :content_md5s, :delete_content, :cocina_path_for, :versions,
+             to: :@service
+
+    def purge_content_files
+      (content_md5s - cocina_content_md5s).each do |md5|
+        delete_content(md5:)
+      end
+    end
+
+    def cocina_content_md5s
+      versions.map do |version|
+        cocina_path = cocina_path_for(version:)
+        raise VersionedFilesService::Error, "Cocina for version #{version} not found" unless cocina_path.exist?
+
+        cocina = Cocina.new(hash: JSON.parse(cocina_path.read))
```

Comment:
Can this be a method in the Cocina class (e.g. `Cocina.for_version(version)`)?

Reply:
Yes. Done.